File: OOP/python/inheritance.py
class Inheritance:
    age = 0
    def method1(self):
        print("Inheritance method1")

    def method2(self):
        print("Inheritance method2")


class App(Inheritance):
    def method1(self):
        print("App method1")



if __name__ == "__main__":
    print("App Started")
    # app = App()
    # app.method1()
    # print(f"Age: {app.age}")
    # app.method2()
    print("App Ended")

File: OOP/c++/inheritance.cpp
#include <iostream>

class Inheritance {  // can be named parent
   public:
    int age;

    Inheritance() {
        this->age = 0;
    }

    // virtual function
    virtual void method1() {
        std::cout << "Inheritance: method1()" << std::endl;
    }
    virtual void method2() {
        std::cout << "Inheritance: method2()" << std::endl;
    }
};

class App : public Inheritance {  // can be named child
   public:
    void method1() {
        std::cout << "App: method1()" << std::endl;
    }
};

int main() {
    std::cout << "App Started" << std::endl;
    App app;
    app.method1();
    std::cout << "Age: " << app.age << std::endl;
    app.method2();
    std::cout << "App Ended" << std::endl;
    return 0;
}